config_templates/pool.yaml

```diff
@@ -129,3 +129,5 @@ pool_specification:
       - kata_containers
       - singularity
     default: null
+  environment_variables:
+    abc: 'xyz'
```

Comment:
There should also be a `keyvault` option here, i.e., `environment_variables_keyvault_secret_id` which will involve a some more effort to get working as a valid keyvault client needs to be passed through fleet into the pool object construction method. If you can't get this working, I can follow up on your PR and add this functionality.